qek/data/graphs.py

```diff
@@ -228,29 +229,63 @@ def compile_register(self) -> pl.Register:
 
     def compile_pulse(
         self,
-        amplitude: float = SEQUENCE_DEFAULT_AMPLITUDE_RAD_PER_US,
-        duration: int = SEQUENCE_DEFAULT_DURATION_NS,
+        amplitude: float | None = None,
+        duration: float | None = None,
     ) -> pl.Pulse:
         """Extract a Pulse for this graph.
 
         A Pulse represents the laser applied to the atoms on the device.
 
         Arguments:
-            amplitude: The amplitude for the laser pulse, in rad per microseconds.
-                By default, use the value demonstrated in the companion paper.
-            duration: The duration of the laser pulse, in nanoseconds.
-                By default, use the value demonstrated in the companion paper.
+            amplitude (optional): The amplitude for the laser pulse, as a value in [0, 1],
+                where 0 is no pulse and 1 is the maximal amplitude for the device. By default,
+                use the value demonstrated in the companion paper.
+            duration (optional): The duration of the laser pulse, as a value in [0, 1],
+                where 0 is the shortest possible duration and 1 is the longest possible
+                duration. By default, use the value demonstrated in the companion paper.
         """
         # Note: In the low-level API, we separate register and pulse compilation for
         # pedagogical reasons, because we want to take the opportunity to teach them
         # about registers and pulses, rather than pulser sequences.
 
-        # See the companion paper for an explanation on these constants.
-        Omega_max = amplitude
-        t_max = duration
+        channel = self.device.channels["rydberg_global"]
+        assert channel is not None
+
+        max_amp = channel.max_amp
+        assert max_amp is not None
+
+        min_duration = channel.min_duration
+        max_duration = channel.max_duration
+        assert max_duration is not None
+
+        if amplitude is None:
+            absolute_amplitude = self.SEQUENCE_DEFAULT_AMPLITUDE_RAD_PER_US
+            if absolute_amplitude > max_amp:
+                # Unlikely, but let's defend in depth.
+                raise ValueError(
+                    f"This device does not support pulses with amplitude {absolute_amplitude} rad per us"
```

Comment:
```suggestion
                    f"This device does not support pulses with amplitude {absolute_amplitude} greater than {max_amp} rad per us"
```

Let's be defensive to the death !